syntax = "proto3";

package nebius.iam.v1;

import "ydb/public/api/client/nc_private/iam/service_account.proto";
import "ydb/public/api/client/nc_private/iam/tenant_user_account.proto";
import "ydb/public/api/client/nc_private/iam/user_account.proto";

option java_multiple_files = true;
option java_outer_classname = "ProfileServiceProto";
option java_package = "ai.nebius.iam.identity.v1";

service ProfileService {
  // option (api_service_name) = "cpl.iam";
  // option (service_visibility) = PUBLIC;
  // option (audit.service_audit).enabled = true;

  rpc Get(GetProfileRequest) returns (GetProfileResponse);
}

message GetProfileRequest{
}

message GetProfileResponse {
  oneof profile{
    UserProfile user_profile = 1;
    ServiceAccountProfile service_account_profile = 2;
    AnonymousAccount anonymous_profile = 3;
  }
}

message ListProfileResponse{
  repeated GetProfileResponse items = 1;
}

message UserProfile {
  string id = 1;
  UserAccountExternalId federation_info = 2;
  oneof attributes_optional {
    UserAttributes attributes = 3;
    Error retrieving_error = 4;
  }
  repeated UserTenantInfo tenants = 5;
}

message UserTenantInfo {
  string tenant_id = 1;
  string tenant_user_account_id = 2;
}

message ServiceAccountProfile {
  ServiceAccount info = 1;
}

message AnonymousAccount {
}
